### PR TITLE
Update VSL.Imports.Targets for signing in Microbuild V2

### PR DESCRIPTION
**Pull request by basoundr**
With this change, We are signing all our present binaries and the single VSIX. In the future, if more binaries gets added to the VSIX then those binaries will be automatically signed as well

But we are shipping a product binary out of a VSIX then Imports.targets needs to updated accordingly

Note: Set the variable RunningInMicroBuild = true in Microbuild definition to sign the product while building the product in microbuildV2

### CODE REVIEW

**Comment by basoundr** [comment]
Tagging @dotnet/roslyn-analysis @davkean @srivatsn @jinujoseph for review

File: build.cmd [hunk 1]
```diff
@@ -33,7 +33,7 @@ set BinariesDirectory=%Root%bin\%BuildConfiguration%\
 set LogFile=%BinariesDirectory%Build.log
 if not exist "%BinariesDirectory%" mkdir "%BinariesDirectory%" || goto :BuildFailed
 
-msbuild /nologo /consoleloggerparameters:Verbosity=minimal /fileLogger /fileloggerparameters:LogFile="%LogFile%";verbosity=detailed /t:"%MSBuildTarget%" /p:Configuration="%BuildConfiguration%" "%Root%build\build.proj" %MSBuildAdditionalArguments%
+msbuild /nologo /nodeReuse:false /consoleloggerparameters:Verbosity=minimal /fileLogger /fileloggerparameters:LogFile="%LogFile%";verbosity=detailed /t:"%MSBuildTarget%" /p:Configuration="%BuildConfiguration%" "%Root%build\build.proj" %MSBuildAdditionalArguments%
```
Comment (davkean) [inline build.cmd]: Can we do this behind a switch so that command-line perf doesn't suck? Did you run into a particular issue that forced this?

File: build/Targets/VSL.Imports.targets [hunk 2]
```diff
@@ -40,8 +40,11 @@
 
   <!-- settings for strong name signing -->
   <PropertyGroup>
-    <RunningInMicroBuild Condition="Exists('$(TF_BUILD_BUILDDIRECTORY)\MicroBuild\MicroBuild.Signing.dll')">true</RunningInMicroBuild>
-    <ShouldSignBuild Condition="'$(RealSignBuild)' == 'true' OR ('$(RunningInMicroBuild)' == 'true' AND '$(SignType)' == 'real')">true</ShouldSignBuild>
+    <MicroBuildSigningLibraryV1>$(TF_BUILD_BUILDDIRECTORY)\MicroBuild\MicroBuild.Signing.dll</MicroBuildSigningLibraryV1>
+    <MicroBuildSigningLibraryV2>$(BUILD_ARTIFACTSTAGINGDIRECTORY)\MicroBuild\Plugins\MicroBuild.Plugins.Signing.1.0.200\build\MicroBuild.Signing.dll</MicroBuildSigningLibraryV2>
```
Comment (davkean) [inline build/Targets/VSL.Imports.targets]: Why do we need v1 and v2?

File: build/Targets/VSL.Imports.targets [hunk 3]
```diff
@@ -391,73 +394,55 @@
...
 
-  <UsingTask TaskName="SignFiles" AssemblyFile="$(TF_BUILD_BUILDDIRECTORY)\MicroBuild\MicroBuild.Signing.dll" Condition="'$(RunningInMicroBuild)' == 'true'" />
+  <UsingTask TaskName="SignFiles" AssemblyFile="$(MicroBuildSigningLibrary)" Condition="'$(ShouldSignBuild)' == 'true'"/>
 
+  <!-- The below properties are set so that the SignVsixInputs Target runs right after we Get the VSIX Source items
+       This allows the signing of the components that go into the VSIX, which gets signed through SignVsix Target-->
   <PropertyGroup>
     <CreateVsixContainerDependsOn>$(CreateVsixContainerDependsOn);SignVsixInputs</CreateVsixContainerDependsOn>
     <PrepareForRunDependsOn>$(PrepareForRunDependsOn);SignVsix</PrepareForRunDependsOn>
-    <ProducingSignedVsix Condition="'$(ShouldSignBuild)' == 'true' AND '$(NonShipping)' != 'true' AND '$(CreateVsixContainer)' == 'true'">true</ProducingSignedVsix>
```
Comment (davkean) [inline build/Targets/VSL.Imports.targets]: Why this change?

File: build/Targets/VSL.Imports.targets [hunk 4]
```diff
@@ -391,73 +394,55 @@
...
 
-    <Message Text="Signing VSIX inputs: using authenticode certificate '%(VsixInputAssembliesToSign.AuthenticodeCertificateName)' for @(VsixInputAssembliesToSign)"/>
+    <Message Text="Signing VSIX inputs: using authenticode certificate Microsoft for %(VsixInputAssembliesToSign.Identity)"/>
 
-    <ItemGroup Condition="'$(RunningInMicroBuild)' == 'true'">
+    <ItemGroup>
       <VsixInputAssembliesToSign>
-        <Authenticode>%(VsixInputAssembliesToSign.AuthenticodeCertificateName)</Authenticode>
+        <Authenticode>Microsoft</Authenticode>
         <StrongName>72</StrongName>
```
Comment (davkean) [inline build/Targets/VSL.Imports.targets]: Is this right? What does is 72 associated with?

File: src/Microsoft.VisualStudio.AppDesigner/Microsoft.VisualStudio.AppDesigner.vbproj [hunk 5]
```diff
@@ -24,7 +24,6 @@
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
     <TargetFrameworkProfile />
     <ProjectSystemLayer>VisualStudioDesigner</ProjectSystemLayer>
-    <StrongNameCertificate>Microsoft</StrongNameCertificate>
```
Comment (davkean) [inline src/Microsoft.VisualStudio.AppDesigner/Microsoft.VisualStudio.AppDesigner.vbproj]: If I'm reading correctly, this is now being signed with MicrosoftShared - which is wrong.

File: src/Microsoft.VisualStudio.Editors/Microsoft.VisualStudio.Editors.vbproj [hunk 6]
```diff
@@ -24,7 +24,6 @@
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
     <TargetFrameworkProfile />
     <ProjectSystemLayer>VisualStudioDesigner</ProjectSystemLayer>
-    <StrongNameCertificate>Microsoft</StrongNameCertificate>
```
Comment (davkean) [inline src/Microsoft.VisualStudio.Editors/Microsoft.VisualStudio.Editors.vbproj]: Ditto.

File: build.cmd (hunk 1 above)
Comment (basoundr) [inline build.cmd]: Yes, during testing the machine got locked down because of this issue. Had to kill the processes to release some resources which made the machine reusable. 

Since this being an in-frequent issue, I will have the default value as `true`.

File: build/Targets/VSL.Imports.targets (hunk 3 above)
Comment (basoundr) [inline build/Targets/VSL.Imports.targets]: We have one VSIX and it is signed. I dont know if we are going to have un-signed VSIX. If we do in the future then we can modify this information. I dont want to overload information where we have a simple scenario to deal with.

File: build/Targets/VSL.Imports.targets (hunk 2 above)
Comment (basoundr) [inline build/Targets/VSL.Imports.targets]: removed V1

**Comment by basoundr** [comment]
ping?

File: build/Targets/VSL.Imports.targets [hunk 7]
```diff
@@ -413,15 +423,16 @@
 
     <!-- The below check is to decide if the assembly produced by the VSIX project should be included in the VSIX-->
     <ItemGroup>
-      <VsixInputAssembliesToSign Include="@(IntermediateAssembly)" Condition="'$(IncludeAssemblyInVSIXContainer)' == 'true'" />
+      <VsixInputAssembliesToSign Include="@(IntermediateAssembly)" Condition="'$(IncludeAssemblyInVSIXContainer)' == 'true'">
+        <StrongName>72</StrongName>
```
Comment (davkean) [inline build/Targets/VSL.Imports.targets]: This doesn't look right, you're not using the results of GetTargetPathWithStrongNameFriendlyID

Comment (basoundr) [inline build/Targets/VSL.Imports.targets]: Currently `ProjectSystem.dll` is not set with the property `SignAssembly=true`. Hence `ProjectSystem.dll` will not enter the condition which assigns `GetTargetPathWithStrongNameFriendlyID`. Will introduce a condition

Comment (basoundr) [inline build/Targets/VSL.Imports.targets]: It was unnecessary to set the value because if the assembly is expected to be part of the VSIX which is going to be signed then all the components will also need to be signed which requires us to set the value of `$SignAssembly` to be true, which will get the assembly assigned with the `$StrongNameCertificateFriendlyId`

**Comment by davkean** [comment]
okay, with the latest commit, it all makes sense now. :+1:
